src/utils/spanUtils.ts

```diff
@@ -0,0 +1,156 @@
+    span.attributes['http.route']
+  ) {
+    tags['ai.operation.name'] = `${span.attributes['http.method']} ${span.attributes['http.route']}`;
+  }
+  return tags;
+}
+
+function createPropertiesFromSpan(span: ReadableSpan): Properties {
+  const properties: Properties = {};
+
+  Object.keys(span.attributes).forEach((key: string) => {
+    if (!(key.startsWith('http.') || key.startsWith('grpc.'))) {
+      properties[key] = span.attributes[key] as string;
+    }
+  });
+
+  const links: MSLink[] = span.links.map((link: Link) => ({
+    operation_Id: link.context.traceId,
+    id: link.context.spanId,
+  }));
+
+  properties['_MS.links'] = links;
+
+  return properties;
+}
+
+function createDependencyData(span: ReadableSpan): RemoteDependencyData {
+  const data = new RemoteDependencyData();
+  data.name = span.name;
+  data.id = `|${span.spanContext.traceId}.${span.spanContext.spanId}.`;
+  data.success = span.status.code === CanonicalCode.OK;
+  data.resultCode = String(span.status.code);
+  data.target = span.attributes['http.url'] as string;
+  data.type = 'HTTP';
+  data.duration = String(hrTimeToMilliseconds(span.duration));
+  data.ver = 1;
+
+  if (span.attributes['http.status_code']) {
+    data.resultCode = String(span.attributes['http.status_code']);
+  }
+
+  if (span.attributes['http.url']) {
+    const url = new URL(span.attributes['http.url'] as string);
+    data.target = url.hostname;
+    data.data = url.href;
+
+    if (span.attributes['http.method']) {
+      data.name = `${span.attributes['http.method']} ${url.pathname}`;
+    }
+  }
+
+  return data;
+}
+
+function createRequestData(span: ReadableSpan): RequestData {
+  const data = new RequestData();
+  data.name = span.name;
+  data.id = `|${span.spanContext.traceId}.${span.spanContext.spanId}.`;
+  data.success = span.status.code === CanonicalCode.OK;
+  data.responseCode = String(span.status.code);
+  data.duration = String(hrTimeToMilliseconds(span.duration));
+  data.ver = 1;
+  data.source = undefined;
+
+  if (span.attributes['http.method']) {
+    data.name = span.attributes['http.method'] as string;
+
+    if (span.attributes['http.status_code']) {
+      data.responseCode = String(span.attributes['http.status_code']);
+    }
+
+    if (span.attributes['http.url']) {
+      data.url = span.attributes['http.url'] as string;
+    }
+
+    if (span.attributes['http.route']) {
+      data.name = `${span.attributes['http.method']} ${span.attributes['http.route']}`;
+    } else if (span.attributes['http.url']) {
+      const url = new URL(span.attributes['http.url'] as string);
+      data.name = `${span.attributes['http.method']} ${url.pathname}`;
+    }
+  }
+
+  // @todo: parse grpc attributes
+
+  return data;
+}
+
+function createInProcData(span: ReadableSpan): RemoteDependencyData {
+  const data = createDependencyData(span);
+  data.type = 'InProc';
+  data.success = true;
+  return data;
+}
+
+export function readableSpanToEnvelope(
```

Comment:
What is actually going to be calling this? AzureMonitorSpanExporter that's going to extend from `AzureMonitorBaseExporter` correct?

Reply:
yes